### DISCUSSION TITLE
Can a cache be shared between different OS versions?

### DISCUSSION THREAD

**Discussion by dyadav7** [ccache/ccache#737] [Questions & answers] [answered]
### Question ###

We are using same NFS ccache on 2 different Centos versions. 

     Host1 $ cat /etc/redhat-release 
     CentOS release 6.5 (Final)
     Host2 $ cat /etc/redhat-release 
     CentOS Linux release 7.8.2003 (Core)

I am seeing some build issues (not sure it is due to ccache). 
Can ccache entries be shared between RH6 and RH7 in any scenario and lead to some issues?

**Comment by jrosdahl** [comment, marked as answer]
It's by default safe to use the same cache directory for different systems, unless the compilers have identical mtime and size (for instance if it's the same compiler wrapper script with the same mtime). See also https://ccache.dev/manual/4.1.html#config_compiler_check and the answer to #725.

**Comment by dyadav7** [comment]
The compiler is also on NFS so, it will have same mtime and size.

**Comment by jrosdahl** [comment]
Then cache entries will be shared as long as source and include files (including system include files) are identical. And if the environment is identical in practice then there shouldn't be any issues.

**Comment by dyadav7** [comment]
Thanks
